src/main/java/seedu/us/among/logic/parser/ParserUtil.java

```diff
@@ -30,9 +31,11 @@
      * @throws ParseException if the specified index is invalid (not non-zero
      *                        unsigned integer).
      */
-    public static Index parseIndex(String oneBasedIndex) throws ParseException {
+    public static Index parseIndex(String oneBasedIndex, String commandUsage) throws ParseException {
         String trimmedIndex = oneBasedIndex.trim();
-        if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
+        if (trimmedIndex.equals("")) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, commandUsage));
+        } else if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
```

Comment:
Interesting that you passed the message into the `parseIndex` method to create a custom error message. I think the alternative is the previous structure which is fine as well, as the exception of an index parsing error need not know which command it is in when the exception occurs. It is when the error is being caught at the higher level, the exception is transformed with added details of the command usage. Let's keep to what you have since this change affects various test cases. Just some thoughts.

Reply:
There... Was a previous method? xD Yea I agree what I did is actually not ideal, given more time (if we're serious about v2.0) we can clean this up in the future. On that note @JulietTeoh if you want to see how the error messages look like on the smallest window size (800 x 600) you may refer to the latest UG sent in our telegram group.